enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import logging
+import os
+import sys
+import time
+import random
+import hashlib
+import zmq
+
+import avalon_enclave_manager.sgx_work_order_request as work_order_request
+import avalon_enclave_manager.avalon_enclave_info as enclave_info
+import avalon_crypto_utils.crypto_utility as crypto_utils
+from avalon_enclave_manager.base_enclave_manager import EnclaveManager
+from avalon_enclave_manager.worker_kv_delegate import WorkerKVDelegate
+from avalon_enclave_manager.work_order_kv_delegate import WorkOrderKVDelegate
+from avalon_enclave_manager.wpe_requester import WPERequester
+from error_code.error_status import ReceiptCreateStatus, WorkOrderStatus
+
+logger = logging.getLogger(__name__)
+
+
+class WorkOrderProcessorEnclaveManager(EnclaveManager):
+    """
```

Comment:
WPE is looks almost similar to singletonEnclave manager except preprocess_work_order call it execute work order. Does it make sense to some function to Enclave manager base class?

Reply:
I am also contemplating on that.